src/test/java/org/springframework/cloud/deployer/spi/kubernetes/DefaultContainerFactoryTest.java

```diff
@@ -69,6 +72,80 @@ public void create() {
+
+		AppDefinition definition = new AppDefinition("app-test", null);
+		Resource resource = getResource();
+		Map<String, String> props = new HashMap<>();
+		props.put("spring.cloud.deployer.kubernetes.containerCommand",
+				"echo, arg1, arg2");
+		AppDeploymentRequest appDeploymentRequest = new AppDeploymentRequest(definition,
+				resource, props);
+
+		Container container = defaultContainerFactory.create("app-test",
+				appDeploymentRequest, null, null);
+		assertNotNull(container);
+		List<String> command = container.getCommand();
+		assertNotNull(command);
+		assertTrue("Command list should contain 3 strings, the command and two arguments", command.size() == 3);
+		assertEquals("echo", command.get(0));
+		assertEquals("arg1", command.get(1));
+		assertEquals("arg2", command.get(2));
+	}
+
+	@Test
+	public void createWithPorts() {
+		KubernetesDeployerProperties kubernetesDeployerProperties = new KubernetesDeployerProperties();
+		DefaultContainerFactory defaultContainerFactory = new DefaultContainerFactory(
+				kubernetesDeployerProperties);
+
+		AppDefinition definition = new AppDefinition("app-test", null);
+		Resource resource = getResource();
+		Map<String, String> props = new HashMap<>();
+		props.put("spring.cloud.deployer.kubernetes.containerPorts",
+				"8081, 8082, 65535");
+		AppDeploymentRequest appDeploymentRequest = new AppDeploymentRequest(definition,
+				resource, props);
+
+		Container container = defaultContainerFactory.create("app-test",
+				appDeploymentRequest, null, null);
+		assertNotNull(container);
+		List<ContainerPort> containerPorts = container.getPorts();
+		assertNotNull(containerPorts);
+		assertTrue("There should be three ports set", containerPorts.size() == 3);
+		assertTrue(8081 == containerPorts.get(0).getContainerPort());
+		assertTrue(8082 == containerPorts.get(1).getContainerPort());
+		assertTrue(65535 == containerPorts.get(2).getContainerPort());
+	}
+
+    @Test
+    public void createWithInvalidPort() {
+        KubernetesDeployerProperties kubernetesDeployerProperties = new KubernetesDeployerProperties();
+        DefaultContainerFactory defaultContainerFactory = new DefaultContainerFactory(
+                kubernetesDeployerProperties);
+
+        AppDefinition definition = new AppDefinition("app-test", null);
+        Resource resource = getResource();
+        Map<String, String> props = new HashMap<>();
+        props.put("spring.cloud.deployer.kubernetes.containerPorts",
+                "8081, 8082, invalid");
+        AppDeploymentRequest appDeploymentRequest = new AppDeploymentRequest(definition,
+                resource, props);
+
+        Container container = defaultContainerFactory.create("app-test",
+                appDeploymentRequest, null, null);
+        assertNotNull(container);
+        List<ContainerPort> containerPorts = container.getPorts();
+        assertNotNull(containerPorts);
+        assertTrue("There should be three ports set", containerPorts.size() == 2);
```

Comment:
Given comment above, this should be changed to assert failure. But this test was copy/pasted and `"three"` was not changed here.